Executable version of the Lean model

ineol opened the pull request:
This adds a new target `generated_lean_rv64d_executable` that does not use axioms nor any non-computable constructs so that it can be used as a library for the forthcoming elf binary runner.

tobiasgrosser reviewed (approved):
Yes, this is super helpful. Thank you, @ineol .

ineol commented:
@jordancarlin  thank you for the feedback! I changed the target name and added it to the CI (but without the lean build because it's very slow at the moment)

ineol commented:
Done!

Timmmm commented on model/CMakeLists.txt:
```diff
@@ -478,28 +501,25 @@ foreach (xlen IN ITEMS 32 64)
+                        ${sail_srcs}
+                )
+                add_custom_command(
+                    DEPENDS ${sail_srcs}
```
This is missing some dependencies - at least `config/default.json` and `handwritten_support/RiscvExtrasExecutable.lean` and `handwritte_supports/RiscvExtras.lean`. It means incremental builds won't work properly if you edit those files. Same for line 497.

Existing problem though so don't feel like you need to fix it in this PR - just pointing it out.

ineol replied:
I fixed it in this PR since it's a quick change. Thank you for the remark!

tobiasgrosser commented on model/CMakeLists.txt:
```diff
@@ -470,6 +470,29 @@ foreach (xlen IN ITEMS 32 64)
+                    --lean-noncomputable-function encdec_compressed_forwards
+                    --lean-noncomputable-function encdec_compressed_backwards
+                    --lean-noncomputable-function encdec_compressed_forwards_matches
+                    --lean-noncomputable-function encdec_compressed_backwards_matches
```
Actually, we do.

tobiasgrosser commented:
@ineol, there seem to be some trailing white spaces that break the CI.
